lib/xnvmec.c

```diff
@@ -1837,7 +1837,7 @@ xnvmec(struct xnvmec *cli, int argc, char **argv, int opts)
 	}
 
 	if ((opts & XNVMEC_INIT_DEV_OPEN) && cli->args.uri) {
-		struct xnvme_opts opts = xnvme_opts_default();
```

Comment:
Why change this? It provided defaults, which would be overwritten by cli-arguments.

Reply:
The problem is that they are overwritten even if the user provides no arguments. All the int values are set to 0

Reply:
I see. ``xnvmec_cli_to_opts`` needs a bit a attention, something like changing:

``opts->be = cli->args.be;``
to
``opts->be = cli->args.given[XNVMEC_OPT_BE] ? cli->args.be : opts->be;``